### PR TITLE
Update the firware binary to work with latest cc3200-rs

### PR DESCRIPTION
**Pull request by fabricedesre**
Changes in this PR are:
- Update to current cc3200-rs master
- src/main.rs is now a simple runner, right now sending temperature readings.
- the wlan setup moved to wlan.rs
- config.rs.sample needs to be copied to config.rs with your actual parameters.

### CODE REVIEW

**Comment by fabricedesre** [comment]
ping @dhylands 
Travis seems to have a hard time with cargo right now...

File: src/config.rs.sample [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+pub const SSID: &'static str = "OpenWireless.org";
+
+pub fn security_params() -> Option<SlSecParams> {
+    // If using an open access point, just return None
+    Some(SlSecParams::wpa2("YOUR-PASSWORD-HERE"))
+    //None
+}
+
+pub const SENSOR_READING_COUNT: u32 = 10;
+pub const SERVER_URL: &'static str = "http://10.252.33.211:8000/endpoint";
```
Comment (dhylands) [inline src/config.rs.sample]: nit: no newline at end of file

File: scripts/travis.sh [hunk 2]
```diff
@@ -9,5 +9,6 @@ export PATH=/home/travis/.cargo/bin:$PATH
 rustup default nightly
 rustup component add rust-src
 cargo install xargo
+cp src/config.rs.sample src/config.rs
```
Comment (dhylands) [inline scripts/travis.sh]: That seems like a reasonable way to do it in cc3200-rs as well (and addresses your concern about checking the file in accidentally).

File: src/main.rs [hunk 3]
```diff
@@ -9,17 +9,72 @@
...
+
+    let i2c = I2C::open(I2COpenMode::MasterModeFst).unwrap();
+    let temp_sensor = TMP006::default(&i2c).unwrap();
+
+    println!("Will now send {} temperature sensing to the server...",
+             config::SENSOR_READING_COUNT);
+
+    for _ in 0..config::SENSOR_READING_COUNT {
+        let temperature: u32 = (temp_sensor.get_temperature().unwrap() * 10.0) as u32;
+        info!("Feels like {}.{} C", temperature / 10, temperature % 10);
```
Comment (dhylands) [inline src/main.rs]: nit: This will report interesting results with a negative temperature :)

File: src/main.rs [hunk 4]
```diff
@@ -9,17 +9,72 @@
...
+            .unwrap()
+            .send(&[])
+            .unwrap()
+            .response(|_| false)
+            .unwrap();
+        let mut buffer = [0u8; 256];
+        info!("Received {}",
+              response.body.read_string_to_end(&mut buffer).unwrap());
+        CurrentTask::delay(Duration::ms(1000))
+    }
```
Comment (dhylands) [inline src/main.rs]: I don't see where this actually sends a temperature to the server.

**Comment by fabricedesre** [comment]
@dhylands I addressed comments and pinned Rustc to the same version as in cc3200-rs. That fixed the unrelated cargo issue which is https://github.com/rust-lang/cargo/issues/3340

**Comment by dhylands** [comment]
lgtm

**Pull request merged**